docs/user-guide/node-api.md

```diff
@@ -77,6 +77,20 @@ If `true`, all disable comments (e.g. `/* stylelint-disable block-no-empty */`)
 
 You can use this option to see what your linting results would be like without those exceptions.
 
+## `cache`
+
+Store the info about processed files in order to only operate on the changed ones. The cache is stored in `.stylelintcache` by default. Enabling this option can dramatically improve Stylelint's running time by ensuring that only changed files are linted.
+
+**Note:** If you run Stylelint with `--cache` and then run Stylelint without `--cache`, the `.stylelintcache` file will be deleted. This is necessary because the results of the lint might change and make `.stylelintcache` invalid. If you want to control when the cache file is deleted, then use `--cache-location` to specify an alternate location for the cache file.
```

Comment:
Actually, cache file will be deleted when `--cache-location` is not changing between runs with `--cache` and without. Cache file won't be deleted for the following case:
1. `stylelint *.scss --cache` - create cache file in default location
1. `stylelint *.scss --cache-location /tmp` - won't delete cache file from default location
1. `stylelint *.scss --cache` - reuse cache file from default location